taiga_halo2/src/transaction.rs

```diff
@@ -201,451 +125,207 @@ impl Transaction {
         Ok(())
     }
 
-    pub fn public_inputs_check(&self) -> Result<(), TransactionError> {
-        for partial_tx in self.partial_txs.iter() {
-            // nullifier check
-            partial_tx.check_nullifiers()?;
-            // output note commitment check
-            partial_tx.check_note_commitments()?;
+    fn get_binding_vk(&self) -> BindingVerificationKey {
+        let mut vk = pallas::Point::identity();
+        if let Some(bundle) = self.shielded_bundle() {
+            vk = bundle
+                .get_value_commitments()
+                .iter()
+                .fold(vk, |acc, cv| acc + cv.inner());
         }
 
-        Ok(())
-    }
+        if let Some(bundle) = self.transparent_bundle() {
+            vk = bundle
+                .get_value_commitments()
+                .iter()
+                .fold(vk, |acc, cv| acc + cv.inner());
+        }
 
-    #[allow(clippy::type_complexity)]
-    pub fn execute(
-        &self,
-    ) -> Result<(Vec<Nullifier>, Vec<NoteCommitment>, Vec<pallas::Base>), TransactionError> {
-        // Verify proofs
-        self.verify_proofs()?;
+        BindingVerificationKey::from(vk)
+    }
 
-        // Verify binding signature
-        self.verify_binding_sig()?;
+    fn digest(&self) -> [u8; 32] {
```

Comment:
If the signature is on a hash of a bundle, then we can't combine differently-signed partial transactions into one bundle.
We probably need to think more about this

Reply:
What do you mean by differently-signed partial transactions?
I wanted to digest the whole tx(including all the transparent and shielded ptxs) and do the binding signature for the balance checking. It's not for authenticity and integrity. The digest contains the notes' nf, cm, value_commitments, and anchor. But I might miss some requisite parameters.

Reply:
Yes
The binding signature is for the final transaction. We can compute/recompute the binding signature whenever the transaction is finalized.